model/artifact/artifact_file_test.go

```diff
@@ -5,82 +5,81 @@ import (
 
 	"github.com/evergreen-ci/evergreen/db"
 	"github.com/evergreen-ci/evergreen/testutil"
-	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/suite"
 	"gopkg.in/mgo.v2/bson"
 )
 
 func init() {
 	db.SetGlobalSessionProvider(testutil.TestConfig().SessionFactory())
 }
 
-func reset(t *testing.T) {
-	testutil.HandleTestingErr(
-		db.Clear(Collection),
-		t, "Error clearing collection")
+type TestArtifactFileSuite struct {
+	testEntry Entry
+	suite.Suite
 }
 
-func TestEntryUpsert(t *testing.T) {
-	Convey("With an artifact file entry", t, func() {
-		reset(t)
+func TestArtifactFile(t *testing.T) {
+	s := &TestArtifactFileSuite{}
+	suite.Run(t, s)
+}
+
+func (s *TestArtifactFileSuite) SetupTest() {
+	s.NoError(db.Clear(Collection))
+	s.testEntry = Entry{
+		TaskId:          "task1",
+		TaskDisplayName: "Task One",
+		BuildId:         "build1",
+		Files: []File{
+			{"cat_pix", "http://placekitten.com/800/600", ""},
+			{"fast_download", "https://fastdl.mongodb.org", ""},
+		},
+		Execution: 1,
```

Comment:
What sets an `Entry`'s execution number in the first place?

Reply:
Search the diff for "$setOnInsert". The first upsert (i.e. insert) will set the execution number.

Reply:
I meant, what sets the execution number in the struct? (not in the db)

Reply:
@bsamek https://github.com/evergreen-ci/evergreen/pull/406/files#diff-4b09415ebcdfdcbfcd1446bf22e978c2R364

Reply:
thanks